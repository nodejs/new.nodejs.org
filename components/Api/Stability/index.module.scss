.stability {
  border-radius: 4px;
  color: #fff;
  line-height: 1.5;
  margin: 0 0 1rem;
  padding: 1rem;

  p {
    display: inline;
    margin: 0;
  }

  a {
    color: #fff;

    code {
      color: #fff !important;
    }
  }

  &Level0 {
    background-color: var(--danger5);
  }

  &Level1 {
    background-color: var(--warning4);
  }

  &Level2 {
    background-color: var(--brand5);
  }

  &Level3 {
    background-color: var(--info5);
  }
}
