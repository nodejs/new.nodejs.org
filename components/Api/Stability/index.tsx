import { FormattedMessage } from 'react-intl';
import styles from './index.module.scss';
import type { PropsWithChildren, FC } from 'react';

type StabilityProps = PropsWithChildren<{ stability: number }>;

const getStabilityClass = (stability: number) => {
  const style = styles[`stabilityLevel${stability}`];
  if (!style) throw new Error(`Unknown stability level: ${stability}`);
  return style;
};

const Stability: FC<StabilityProps> = ({ stability, children }) => (
  <div className={`${styles.stability} ${getStabilityClass(stability)}`}>
    <FormattedMessage
      id="components.api.stability"
      values={{ level: stability }}
    />
    {children}
  </div>
);

export default Stability;
