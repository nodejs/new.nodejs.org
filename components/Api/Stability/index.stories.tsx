import Stability from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof Stability>;
type Meta = MetaObj<typeof Stability>;

export const Default: Story = {
  args: {
    stability: 0,
    children: 'This is an stability index example',
  },
  argTypes: {
    stability: {
      control: {
        type: 'range',
        min: 0,
        max: 3,
        step: 1,
      },
    },
  },
};

export default { component: Stability } as Meta;
