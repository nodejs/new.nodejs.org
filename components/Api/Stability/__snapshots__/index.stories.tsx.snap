// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api/Stability Default smoke-test 1`] = `
<div class="Stability_stability__U0Zm0 Stability_stabilityLevel0__ygvXd">
  Stability: 0 - This is an stability index example
</div>
`;
